src/azstoragetorch/io.py

```diff
@@ -153,8 +158,58 @@ def _blob_url_has_sas_token(self, blob_url: str) -> bool:
         # key to determine if the URL has a SAS token.
         return "sig" in parsed_qs
 
+    def _readline(self, size: Optional[int]) -> bytes:
+        consumed = b""
+        if size == 0 or self._is_at_end_of_blob():
+            return consumed
+
+        limit = self._get_limit(size)
+        if self._readline_buffer:
+            consumed = self._consume_from_readline_buffer(consumed, limit)
+        while self._should_download_more_for_readline(consumed, limit):
+            self._readline_buffer = self._client.download(
+                offset=self._position, length=self._READLINE_PREFETCH_SIZE
+            )
+            consumed = self._consume_from_readline_buffer(consumed, limit)
+        return consumed
+
+    def _get_limit(self, size: Optional[int]) -> Optional[int]:
+        limit = size
+        if size is not None and size < 0:
+            limit = None
+        return limit
```

Comment:
Not sure how you'll feel about this but it may be easier to use `sys.maxsize` instead of `None` if there is no limit? It just means the return type can't be `None` which makes working with it a little easier later on.

Reply:
I definitely agree it would make the logic simpler by reducing branching. I'm hesitant to use `sys.maxsize` because while Blobs do not support blobs of that size, it does not mean it won't in the future. Instead, I'm thinking we could use the blob size instead as the limit, which is already available on the client and we grab and cache anyways on downloads.